src/keys/slip10.rs

```diff
@@ -99,384 +188,196 @@ impl Seed {
         Self(bs.to_vec())
     }
 
-    pub fn to_master_key(&self, curve: Curve) -> ExtendedSecretKey {
-        ExtendedSecretKey::from_seed(curve, self)
+    pub fn to_master_key<K: hazmat::IsSecretKey>(&self) -> Extended<K> {
+        Extended::from_seed(self)
     }
 
-    pub fn derive(&self, curve: Curve, chain: &Chain) -> crate::Result<ExtendedSecretKey> {
-        self.to_master_key(curve).derive(chain)
+    pub fn derive<K: hazmat::IsSecretKey>(&self, chain: &Chain) -> crate::Result<Extended<K>> {
+        self.to_master_key().derive(chain)
     }
 }
 
 pub type ChainCode = [u8; 32];
 
-#[derive(Clone, Zeroize, ZeroizeOnDrop)]
-pub struct ExtendedSecretKey(KeyImpl);
-
-impl ExtendedSecretKey {
-    pub fn from_seed(curve: Curve, seed: &Seed) -> Self {
-        Self(KeyImpl::from_seed(curve, &seed.0))
-    }
+#[derive(ZeroizeOnDrop)]
+pub struct Extended<K> {
```

Comment:
Is `Extended` a descriptive enough name?

Reply:
The idea was it'll be used with concrete types, eg. `Extended<ed25519::SecretKey>`, `Extended<secp256k1_ecdsa::PublicKey>`. No need to duplicate "Key" twice. I can still change to `ExtendedKey` if you insist.

Reply:
Renamed to `Slip10`.